enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,408 @@
+  To mitigate this we could add that validation, at least on installer, to provide quick-feedback (fail when validate install-config) to the user when the provided EIP is already associated to another resource.
+  It would be nice to have a validation before setting the annotation to CCM, keeping the operator degraded before disrupting the service.
+
+## Test Plan
+
+### Ingress Operator Testing
+
+An E2E test will be created in the Ingress Operator repo to verify the functionality of the new [IngressConfig API](#ingressconfig-api-updates).
+This test will follow a similar pattern established in the existing [TestAWSLBTypeDefaulting](https://github.com/openshift/cluster-ingress-operator/blob/4e621359cea8ef2ae8497101ee3daf9f474b4b66/test/e2e/operator_test.go#L1368) test.
+
+### Installer Testing
+
+E2E test(s) will also be added to the installer to verify functionality of the new [Installer API](#installer-updates).
+These tests, typically written by QE, will follow existing patterns established for testing installer functionality in
+the [openshift-tests-private](https://github.com/openshift/release/tree/master/ci-operator/config/openshift/openshift-tests-private)
+directory of the openshift/release repo. These installer tests are run as nightly CI jobs.
+
+
+## Graduation Criteria
+
+This feature will initially be released as Tech Preview only, behind the `TechPreviewNoUpgrade` feature gate.
+
+### Dev Preview -> Tech Preview
+
+N/A. This feature will be introduced as Tech Preview.
+
+### Tech Preview -> GA
+
+The E2E tests should be consistently passing and a PR will be created
+to enable the feature gate by default.
+
+### Removing a deprecated feature
+
+## Upgrade / Downgrade Strategy
+
+No upgrade or downgrade concerns because all changes are compatible or in the installer.
+
+## Version Skew Strategy
+
+N/A. This enhancement will only be supported for new installations and therefore has no version skew concerns.
+
+## Operational Aspects of API Extensions
+
+N/A.
+
+## Support Procedures
+
+CCM - Cloud  Controller Manager
+
+- If the service is not getting updated with the EIP annotation and the kubernetes load balancer
+  service is in `Pending` state then check the `cloud-controller-manager` logs and `IngressController` status.
+
+In the CCM logs search for the `IngressController` name and check if you find any error related EIP allocation failure.
+```sh
+  oc logs <ccm pod name> -n openshift-cloud-controller-manager
+```
+
+In the `IngressController` status, check the status for the following:
+- Error messages for invalid eips or eips not present in the subnet of the VPC is `The service-controller component is reporting SyncLoadBalancerFailed events like: Error syncing load balancer: failed to ensure load balancer: error creating load balancer: "AllocationIdNotFound:` for status type `LoadBalancerReady` and `Available`.
```

Comment:
When this error occurs, does the Installation fail or can the user edit the `ingressNetworkLoadBalancer` list within the Ingress Config spec so that CCM can successfully reconcile the creation of the Ingress LB?

Reply:
@sadasu can you please check  point 3 under section ### Implementation Details/Notes/Constraints

Please let me know WYT.

Reply:
Also if an error is invoked we can either stop or ask for the correct input from user. i need to check the installer if we can add a loop.

Reply:
> @sadasu can you please check point 3 under section ### Implementation Details/Notes/Constraints
> 
> Please let me know WYT.

@miheer OK with that section.
Probably out-of-scope for this enhancement, when Installer determines the predicted LB subnet count, would it useful for the Installer to pass the predicted LB subnets to the AWS CCM via a manifest? That way we will not be duplicating this logic of figuring out the correct subnets to use. /cc @patrickdillon @mtulio